Classes/New Issue/IssueTemplates.swift

```diff
@@ -0,0 +1,220 @@
+        ) {
+
+        self.fetchFile(
+            owner: owner,
+            repo: repo,
+            branch: "master",
+            path: ".github/ISSUE_TEMPLATE/\(filename)") { (result) in
+                switch result {
+                case .success(let file):
+                    completion(.success(file))
+                case .error(let error):
+                    completion(.error(error))
+                case .nonUTF8:
+                    completion(.error(nil))
+                }
+        }
+    }
+
+    private func createTemplateWith(
+        owner: String,
+        repo: String,
+        filename: String,
+        completion: @escaping (Result<IssueTemplate>) -> Void
+        ) {
+        fetchTemplateFile(owner: owner, repo: repo, filename: filename) { (result) in
+            switch result {
+            case .success(let file):
+                let nameAndDescription = IssueTemplateHelper.getNameAndDescriptionFromTemplateFile(file: file)
+                if let name = nameAndDescription.name {
+                    completion(.success(IssueTemplate(title: name, template: file)))
+                } else {
+                    completion(.error(nil))
+                }
+            case .error(let error):
+                completion(.error(error))
+            }
+        }
+    }
+
+    func createNewIssue(
+        owner: String,
+        repo: String,
+        session: GitHubUserSession?,
+        mainViewController: UIViewController,
+        delegate: NewIssueTableViewControllerDelegate) {
+
+        var templates: [IssueTemplate] = []
+        let templateGroup = DispatchGroup()
+
+        self.fetchFiles(
+            owner: owner,
+            repo: repo,
+            branch: "master",
+            path: ".github/ISSUE_TEMPLATE") { (result) in
+                switch result {
+                case .success(let files):
+                    for file in files {
+                        templateGroup.enter()
+                        self.createTemplateWith(owner: owner, repo: repo, filename: file.name, completion: {
+                            switch $0 {
+                            case .success(let template):
+                                templates.append(template)
+                            case .error(let error):
+                                if let err = error {
+                                    Squawk.show(error: err)
+                                }
+                            }
+                            templateGroup.leave()
+                        })
+                    }
+                case .error(let error):
+                    Squawk.show(error: error)
+                }
+
+                // Wait for async calls in for-loop to finish up
+                templateGroup.notify(queue: .main) {
+
+                    // Sort lexicographically
+                    let sortedTemplates = templates.sorted(by: {$0.title < $1.title })
+
+                    if sortedTemplates.count > 0 {
```

Comment:
This logic can be pulled out, for instance.

Reply:
Pulled it out, is it better?